libvirt/image/qemu-linux.pkr.hcl

```diff
@@ -0,0 +1,127 @@
+  format            = "qcow2"
+  headless          = true
+  iso_checksum      = "${var.cloud_image_checksum}"
+  iso_url           = "${var.cloud_image_url}"
+  output_directory  = "output"
+  qemuargs          = [["-m", "${var.memory}"], ["-smp", "cpus=${var.cpus}"], ["-cdrom", "${var.cloud_init_image}"], ["-serial", "mon:stdio"]]
+  ssh_password      = "${var.ssh_password}"
+  ssh_port          = 22
+  ssh_username      = "${var.ssh_username}"
+  ssh_wait_timeout  = "300s"
+  vm_name           = "${var.qemu_image_name}"
+  shutdown_command  = "sudo shutdown -h now"
+}
+
+source "qemu" "rhel" {
+  boot_command      = ["<enter>"]
+  disk_compression  = true
+  disk_image        = true
+  disk_size         = "${var.disk_size}"
+  format            = "qcow2"
+  headless          = true
+  iso_checksum      = "${var.cloud_image_checksum}"
+  iso_url           = "${var.cloud_image_url}"
+  output_directory  = "output"
+  qemuargs          = [["-m", "${var.memory}"], ["-smp", "cpus=${var.cpus}"], ["-cdrom", "${var.cloud_init_image}"], ["-serial", "mon:stdio"], ["-cpu", "Cascadelake-Server"]]
+  ssh_password      = "${var.ssh_password}"
+  ssh_port          = 22
+  ssh_username      = "${var.ssh_username}"
+  ssh_wait_timeout  = "300s"
+  vm_name           = "${var.qemu_image_name}"
+  shutdown_command  = "sudo shutdown -h now" 
+}
+
+
+build {
+  sources = ["source.qemu.ubuntu"]
+
+  provisioner "shell-local" {
+    command = "tar cf toupload/files.tar files"
+  }
+
+  provisioner "file" {
+    source      = "./toupload"
+    destination = "/tmp/"
+  }
+
+  provisioner "shell" {
+    inline = [
+      "cd /tmp && tar xf toupload/files.tar",
+      "rm toupload/files.tar"
+    ]
+  }
+
+  provisioner "file" {
+    source      = "copy-files.sh"
+    destination = "~/copy-files.sh"
+  }
+
+  provisioner "shell" {
+    remote_folder = "~"
+    inline = [
+      "sudo bash ~/copy-files.sh"
+    ]
+  }
+
+  provisioner "file" {
+    source      = "misc-settings.sh"
+    destination = "~/misc-settings.sh"
+  }
+
+  provisioner "shell" {
+    remote_folder = "~"
+    inline = [
+      "sudo bash ~/misc-settings.sh"
+    ]
+  }
+
+}
+
+
+build {
+  sources = ["source.qemu.rhel"]
+
+  provisioner "shell-local" {
+    command = "tar cf toupload/files.tar files"
+  }
+
+  provisioner "file" {
+    source      = "./toupload"
+    destination = "/tmp/"
+  }
+
+  provisioner "shell" {
+    inline = [
+      "cd /tmp && tar xf toupload/files.tar",
+      "rm toupload/files.tar"
+    ]
+  }
+
+  provisioner "file" {
+    source      = "copy-files-rhel.sh"
+    destination = "~/copy-files-rhel.sh"
+  }
+
+  provisioner "shell" {
+    remote_folder = "~"
+    inline = [
+      "sudo bash ~/copy-files-rhel.sh"
+    ]
+  }
+  provisioner "file" {
+    source      = "misc-settings.sh"
+    destination = "~/misc-settings.sh"
+  }
+
+  provisioner "shell" {
+    remote_folder = "~"
+    inline = [
+      "sudo bash ~/misc-settings.sh"
+    ]
+  }
+}
```

Comment:
I think it's better to have a separate packer file for centos/rhel

Reply:
@bpradipt  From my understanding, there are a few restrictions here. You can only specify a specific template file if your variables are also defined in the same file. So, everything in variable*.hcl has to move to both qemu-ubuntu*.hcl and qemu-rhel*.hcl.  The other option is to create separate directories for ubuntu and rhel and copy over the variables file to both. Having both builders in the same file seems to be the convenient option here (since variables are common) but please let me know what you think.

Reply:
Is there a need to have different variable*.hcl file ? Can't we have a single variable file and instantiate the required values in the Makefile ? If not, then putting it in separate directories makes sense.

Reply:
You can also explicitly specify the var file in the CLI - https://developer.hashicorp.com/packer/guides/hcl/variables
Otoh, having separate directories might be cleaner.  

It'll be good to discuss the different approaches in the related issue or over slack as the change impacts all the providers

Reply:
If you are referring to the "-var-file" switch, I think that is meant for variable assignments  not definitions.  Either way, tbh, I was just trying out with separate dirs (rhel/ubuntu) and it isn't that bad. The build just works without any modifications.